test/fail_compilation/fail_shared.d

```diff
@@ -0,0 +1,24 @@
+/*
+REQUIRED_ARGS:-preview=restrictiveshared
+TEST_OUTPUT:
+---
+fail_compilation/fail_shared.d(15): Error: Trying to Access shared state `x`
+fail_compilation/fail_shared.d(16): Error: Trying to Access shared state `x`
+fail_compilation/fail_shared.d(17): Error: Trying to Access shared state `x`
+fail_compilation/fail_shared.d(22): Error: Trying to Access shared state `x`
+---
+*/
+
+
+int f1(shared int x)
+{
+    x += 7; // Error: Trying to accsess shared state x
+    x = 22; // Error: Trying to accsess shared state x
+    return x; // Error: Trying to accsess shared state x
+}
+
+int sync(shared int *x, int y)
+{
+    return x = y; // Error: Trying to accsess shared state x
```

Comment:
integers cannot be assigned to pointers anyway

Reply:
auto-deref.

Reply:
if (e2.type == e1.type.nextof) *e1 = e2

Reply:
> auto-deref.

You said that the other day, but I don't get it...
```D
    int y = 10;
    int* x;
    x = y;
```
```
error : cannot implicitly convert expression `y` of type `int` to `int*`
```